### PR TITLE
Product benchmark

### PR DESCRIPTION
**Pull request by jgosmann**
PR #658 against `master` this time.

### CODE REVIEW

File: nengo/networks/tests/test_product.py [hunk 1]
```diff
@@ -35,3 +35,100 @@ def test_sine_waves(Simulator, plt, seed):
     plt.xlim(right=t[-1])
 
     assert rmse(AB[:len(offset), :], sim.data[p][offset, :]) < 0.2
+
+
+class HilbertCurve(object):
```
Comment (hunse) [inline nengo/networks/tests/test_product.py]: Could we want this for other things? Is it worth putting it somewhere else?

Comment (jgosmann) [inline nengo/networks/tests/test_product.py]: I don't know ... do we need a space filling curve somewhere else?

Comment (hunse) [inline nengo/networks/tests/test_product.py]: I'm not sure if we need it right now, but I could see it being useful for other tests. Also, it seems kind of strange or at least surprising having it right in the middle of this test file.

Comment (jgosmann) [inline nengo/networks/tests/test_product.py]: Yeah, any suggestion where to put it? `nengo.utils.weird_things_that_do_not_belong_anywhere_else`? :wink: 
Maybe `nengo.utils.functions`?

Comment (hunse) [inline nengo/networks/tests/test_product.py]: Yeah, it's being used as an input function, so `nengo.utils.functions` seems appropriate.

**Comment by jgosmann** [comment]
Moved and documented `HilbertCurve`, fixed an error while doing it.

**Comment by hunse** [comment]
Why is this marked as "needs changes"?

**Comment by Seanny123** [comment]
@hunse I think I meant to mark it as `needs rebase` and then derped. Sorry.

**Comment by hunse** [comment]
Rebased and squashed this. Also added a commit to remove the convoluted seed generation that got in there somehow. If @jgosmann is good with this, I think it's good to go.

**Comment by tbekolay** [comment]
LGTM!

**Comment by jgosmann** [comment]
Doesn't look good to me. Now all the trials have the same seed, but they should have different seeds.

**Comment by hunse** [comment]
Oh, I see, you wanted something like this.

**Comment by hunse** [comment]
All I changed now is to just get an rng rather than get a seed to make an rng.

**Comment by jgosmann** [comment]
LGTM :) :cake:

**Comment by jgosmann** [comment]
Won't need a changelog entry, will it?

**Pull request merged**